lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -71,11 +71,8 @@ def serializable_hash_for_single_resource(options)
         end
 
         def resource_identifier_type_for(serializer)
-          if ActiveModel::Serializer.config.jsonapi_resource_type == :singular
-            serializer.object.class.model_name.singular
-          else
-            serializer.object.class.model_name.plural
-          end
+          type = serializer.object.class.name.demodulize.underscore
```

Comment:
For the moment, I would argue this is incorrect.  e.g `User::Mailer`, `Admin::Mailer` would both resolve to mailer. and see https://github.com/rails-api/active_model_serializers/pull/1029#discussion_r37211368